xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go

```diff
@@ -138,9 +138,12 @@ func registerWrappedCDSPolicyWithNewSubConnOverride(t *testing.T, ch chan *xdscr
 func setupForSecurityTests(t *testing.T, bootstrapContents []byte, clientCreds, serverCreds credentials.TransportCredentials) (*grpc.ClientConn, string) {
 	t.Helper()
 
-	xdsClient, xdsClose, err := xdsclient.NewForTesting(xdsclient.OptionsForTesting{
-		Name:     t.Name(),
-		Contents: bootstrapContents,
+	pool, err := xdsclient.NewPool(bootstrapContents)
```

Comment:
Instead of having to change all callsites of `xdsclient.NewForTesting` with a call to create a pool followed by creating a client inside that pool, you could consider adding a helper function in the `testutils` package that accepts bootstrap contents and `xdsclient.OptionsForTesting`, and in turn, creates a new pool and create a new client within that pool and returns the newly created client, the client close func and an error.

Reply:
I have removed `xdsclient.NewForTesting`. The pool have a method `NewClientForTesting` which is updated to be used in the tests.